### ISSUE TITLE
Unable to embed react fragment within Drawer Navigator

### ISSUE THREAD

**Issue by NathanC** [react-navigation/react-navigation#10564]
### Current behavior

Including a react fragment within a Drawer Navigator leads to this error message:

```
A navigator can only contain 'Screen', 'Group' or 'React.Fragment' as its direct children (found 'undefined'). 
To render this component in the navigator, pass it in the 'component' prop to 'Screen'.
```

In other words, the error says that React Fragments are allowed as children, but the error gets thrown even when the child is a React Fragment.

### Expected behavior

Expected the React Fragment to be rendered on top of the screen that gets rendered.

The purpose of doing this is to have a custom header that shows up on each page, and has some components within the header.

### Reproduction

https://snack.expo.dev/@nathantjcc/react-fragment-within-drawer-navigator

### Platform

- [X] Android
- [ ] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/​bottom-tabs
- [X] @react-navigation/​drawer
- [ ] @react-navigation/​material-bottom-tabs
- [ ] @react-navigation/​material-top-tabs
- [ ] @react-navigation/​stack
- [ ] @react-navigation/​native-stack

### Environment

- [x] I've removed the packages that I don't use

    "expo-constants": "~13.0.2",
    "@expo/vector-icons": "^12.0.0",
    "react-native-paper": "4.9.2",
    "react-native-screens": "~3.10.1",
    "react-native-reanimated": "~2.3.1",
    "@react-navigation/drawer": "^6.4.1",
    "@react-navigation/native": "6.0.0",
    "react-native-gesture-handler": "~2.1.0",
    "react-native-safe-area-context": "3.3.2"

**Comment by NathanC** [comment]
I've updated my reproduction with `6.0.10`, Mr.Bot, and the issue is still there.

**Comment by satya164** [comment]
This is not a bug, it's the expected behavior as documented. `Drawer.Navigator`'s children are configuration, they don't get rendered.

**Comment by NathanC** [comment]
@satya164 

Thank you for your quick response! Good to know that this is intended behavior, although the error message is very misleading, so it might be worth changing the wording on it.

Is there a way I can add custom components that get rendered above/below each screen in `@react-navigation/drawer`?

I updated my reproduction here: https://snack.expo.dev/@nathantjcc/custom-header-for-drawer-navigator

I created a hamburger menu outside of the React Drawer component, but I'd like the side menu to go on top of the custom header (instead of the current behavior which is going beneath the header). I'd also like to access the drawer open/closed state (so I can change the hamburger menu icon), but I can't since the Header isn't a child of the Drawer Navigator.

Any suggestions? I've been hacking at this for a while.
